pkg/kubelet/dockershim/network/cni/cni.go

```diff
@@ -435,3 +460,33 @@ func (plugin *cniNetworkPlugin) buildCNIRuntimeConf(podName string, podNs string
 
 	return rt, nil
 }
+
+// addAnnotationsToNetConfList will add annotations as label args to all the network configs. The cni plugins
+// can choose to ignore these args or use them for custom functionality.
+func addAnnotationsToNetConfList(netConfList *libcni.NetworkConfigList, annotations map[string]string) error {
+	cniLabels := make([]cniLabel, 0, len(annotations))
+	for key, value := range annotations {
+		cniLabels = append(cniLabels, cniLabel{
+			Key:   key,
+			Value: value,
+		})
+	}
+
+	inject := map[string]interface{}{
+		"args": cniArgs{
+			CNI: cniArg{
```

Comment:
Annotations are kubernetes-specific. We don't want to put them in the standard `args.cni` field. You should add a new field like `args.k8sAnnotations`.

However, I'm also realizing that any change here will only affect people using docker, not people using other CRI runtimes. We would need some change to the CRI spec or something to make this generically available. And then all the CRI runtimes would have to implement it. Which would mean that plugins would not be able to depend on this feature existing for quite some time. So perhaps it doesn't really make sense to implement this...

Reply:
I had kept them under `args.cni.labels` which is what the CNI said to pass generic key value metadata.

Yes, this change would only affect people using docker and i would make change to the containerd also to support this. This can't go into the CRI spec because this is metadata the CRI want's to pass to the CNI plugins. Even currently pod name and namespace name passed to the CNI are not part of any spec but something which kubernetes implementation is passing right. I also saw the containerd and saw that they have separate repo (containerd/cri) to integrate with kubernetes separate from the generic containerd repo.
let me know if my understanding is wrong as i am new to these repo's.